src/ARToolkit/ARSceneView.cs

```diff
@@ -156,8 +156,14 @@ private void LocationDataSource_LocationChanged(object sender, Location.Location
                 var locationPoint = e.Position;
                 if (_locationTrackingMode == ARLocationTrackingMode.Initial || !_initialLocationSet)
                 {
+#if __ANDROID__
+                    double heading = _initialHeading ?? 0;
```

Comment:
I think this should only be done if `NorthAlign` and `IsUsingARCore` is true. ie
`double heading = NorthAlign && IsUsingARCore && _initialHeading.HasValue ? _initialHeading.Value : 0;`
In addition I'm still not entirely convinced we should be using the initial value, but rather the current value of the compass heading. Or rather it should be the value at the point in time where ARCore started up as started tracking, as that is what ARCore considers "0"
An even simpler solution might be to just grab the heading from the current OriginCamera.
ie `var heading = OriginCamera.Heading;`

Reply:
I updated pull request to use heading from OriginCamera instead.
Seems to also work well when I tested it.